### PR TITLE
Improve handling and display of reload times

### PR DESCRIPTION
**Pull request by mugling**
Summary of behavioral changes:
- **Scale magazine reload times proportional to rounds actually loaded** (16f5277, e8acb35)
  Reloading a 30-round STANAG magazine should take 3 minutes without any practice and just over 90 seconds at `marksmanship 10`. This excludes time fetching ammo from inventory or map.
- **Improves scaling of times when obtaining items via splitting stacks** (a5fb7cb)
  A previous PR prevented `item_location::obtain` from obtaining excessive quantities of ammo. This PR now ensures the correct (lesser) amount of moves is consumed when splitting stacks.
- **Drops `range` and `accuracy` fields from reload prompt** (56ddb34)
  These didn't include the effects of the gun and removal is desirable in keeping the size of the reload prompt manageable. The `damage` and `pierce` fields remain and a future PR will update these to be the sum of the values for gun and ammo.
- **Display raw move costs for reloading** (b795b54, a5fb7cb)
  A future PR may replace these with descriptive text, eg `immediate`, `quick`, `slow` etc but for now the values themselves are useful for verifying intended behavior.

In descending order of speed:
- 1. Reloading using magazine from worn ammo pouch
- 2. Reloading using magazine from inventory
- 3. Reloading using magazine from ground
- 4. Reloading using magazine from adjacent tile
- 5. Reloading a magazine using ammo

Future work:
- Move `pick_reload_ammo` in to `player` and return `target`, `ammo` and `qty` to caller
- Extend `player::item_handling_costs` to accept a `qty` parameter to handle stacks

### CODE REVIEW

File: src/item_location.cpp [hunk 1]
```diff
@@ -173,6 +160,48 @@ class item_location::item_on_person : public item_location::impl
...
+            if( !parents.empty() && who.is_worn( *parents.back() ) ) {
+                // if outermost parent item is worn status effects (eg. GRABBED) are not applied
+                // holsters may also adjust the volume cost factor
+
+                if( parents.back()->can_holster( obj, true ) ) {
+                    auto ptr = dynamic_cast<const holster_actor *>
+                               ( parents.back()->type->get_use( "holster" )->get_actor_ptr() );
+                    mv += dynamic_cast<player &>( who ).item_handling_cost( obj, false, ptr->draw_cost );
+
+                } else {
```
Comment (mugling) [inline src/item_location.cpp]: future PR needs to extend bandoliers to have an equivalent to `draw_cost`.

**Comment by mugling** [comment]
8eb9236 sorts menu by move cost, then remaining ammo. Empty magazines always sort last

**Comment by mugling** [comment]
Done for now

**Comment by mugling** [comment]
Done for now

**Comment by Night-Pryanik** [comment]
>Reloading a 30-round STANAG magazine should take 3 minutes without any practice and just over 90 seconds at marksmanship 10.

I think this is more a matter of skill in particular gun type (rifles, smg and so on) rather than `marksmanship`. Also, it could depend on dexterity too IMO.

**Comment by kevingranade** [comment]
Are a rifle and a SMG going to have a fundamentally different magazine type?  Don't get too caught up on the name, marksmanship is any general gun-handling skill, and rifles/smg/pistol/etc are skills that are specific to that class of gun.

**Comment by chaosvolt** [comment]
It might be acceptable to leave it at the marksmanship level, as @kevingranade suggested. Plus if we wanted to get specific, it would be more realistic to go full gun nerd and base it off a user's familiarity with a given gun and related methods of operation.

Cue glorious Soviet Survivors tugging on the magazine wondering why they can't find the mag release, and 'murrican Survivors futilely poking around just behind an AK's rear sight.

**Comment by mugling** [comment]
6cb042d gets the tests compiling again. I'm going to update and extend these as part of a later PR

**Comment by Night-Pryanik** [comment]
@kevingranade Following your logic - are a rifle and a SMG going to have a fundamentally different gun handling mechanics? Especially some fully-automatic carbine rifle and SMG.

**Comment by kevingranade** [comment]
There are certainly going to be some on the line between one class and
another, but yes you handle a SMG and a rifle differently.
I don't see what that has to do with which skill governs magazine reload
speed though.

**Comment by narc0tiq** [comment]
Interesting -- all the `g++`es that are compiling the code on Jenkins' side are complaining:
```
ccache /home/narc/mxe/usr/bin/i686-w64-mingw32.static-g++  -DRELEASE -DCROSS_LINUX -DGIT_VERSION -DLOCALIZE -ffast-math -Os -Wall -Wextra   -Werror --std=c++11 -MMD -DLUA -c src/item.cpp -o objwin/item.o
src/item.cpp: In lambda function:
src/item.cpp:4249:48: error: suggest parentheses around comparison in operand of '!=' [-Werror=parentheses]
         return lhs.ammo->ammo_remaining() != 0 > rhs.ammo->ammo_remaining() != 0;
                                                ^
src/item.cpp:4249:43: error: suggest parentheses around comparison in operand of '!=' [-Werror=parentheses]
         return lhs.ammo->ammo_remaining() != 0 > rhs.ammo->ammo_remaining() != 0;
                                           ^
cc1plus: all warnings being treated as errors
make: *** [objwin/item.o] Error 1
```
(via <http://ci.narc.ro/job/Cataclysm-Matrix/4459/Graphics=Curses,Platform=Windows/console>)

The `clang++`s (Linux-Curses and Linux-Tiles) are perfectly fine with it. Personally, I'm inclined in favour of the g++es, that just looks confusing as hell.

**Comment by Night-Pryanik** [comment]
I meant that if there is differentiation between rifles and smg in terms of handling and firing, there should also be differentiation between them in terms of reloading.

**Comment by kevingranade** [comment]
Loading the magazines into the guns, maybe, loading rounds into magazines,
no.
That doesn't follow anyway, some aspects of gun handling can be specific to
the type of gun and others can be generic.

**Comment by Night-Pryanik** [comment]
I wonder if it add fun or interest to the game, if we have differentiation not between gun type (rifle or smg), but between type of action - single/double action, break-action, bolt action, lever-action, pump-action an so on.

**Comment by kevingranade** [comment]
You've gotten quite off topic, if you have something specific to suggest
please open another issue, if you want to talk about gun type
differentiation in general please open a thread on the forums.

**Comment by Night-Pryanik** [comment]
Yeah, you're right, that was mostly my thoughts to myself. I'm shutting up.

**Comment by kevingranade** [comment]
The main thing I want to avoid is losing the conversation when the pr is
closed :)

**Pull request merged**